### PR TITLE
Change index expression's start position

### LINKED ISSUE #718: Change indexing expression start position to
### Problem

For historic reasons, index expressions' start position is currently the opening square bracket and not the indexed expression's start position.

This leads to incorrect error ranges and thus confusing errors for users.

### Definition of Done

Change the index expression AST node:
- [ ] Add `IndexingStartPos` field to indicate the position of the opening square bracket
- [ ] Change the start position to the start position of the indexed expression

### PR DESCRIPTION
**Pull request by RZhang05**
Closes #718

## Description

This PR adds a test which throws an error involving index expression different from the type mismatch error addressed in #719. Index expression's start position was changed from opening square bracket to indexed expression's start position.

______

- [x] Targeted PR against `master` branch
- [x] Linked to Github issue with discussion and accepted design OR link to spec that describes this work
- [x] Code follows the [standards mentioned here](https://github.com/onflow/cadence/blob/master/CONTRIBUTING.md#styleguides)
- [x] Updated relevant documentation 
- [x] Re-reviewed `Files changed` in the Github PR explorer
- [x] Added appropriate labels

### CODE REVIEW

File: runtime/parser/expression.go [hunk 1]
```diff
@@ -1262,7 +1262,7 @@ func defineIndexExpression() {
 				firstIndexExpr,
 				ast.NewRange(
 					p.memoryGauge,
-					token.StartPos,
+					left.StartPosition(),
```
Comment (SupunS) [inline runtime/parser/expression.go:1265]: now that `token` parameter is not used anymore, we can omit declaring this unused parameter by using the underscore (`_`) as the parameter name.
i.e: in the function signature above:
```go
func(p *parser, _ lexer.Token, left ast.Expression) (ast.Expression, error) {
```

File: runtime/account_test.go [hunk 2]
```diff
@@ -2676,6 +2676,49 @@ func TestRuntimeGetAuthAccount(t *testing.T) {
 
 		assert.IsType(t, &sema.NotDeclaredError{}, errs[0])
 	})
+
+	t.Run("update names index expression start position", func(t *testing.T) {
```
Comment (SupunS) [inline runtime/account_test.go:2680]: Nice work adding this test!

Tests in the `runtime` package are usually "integration tests" in a way. As in, it setup the environments, create transactions, etc. For this change, given we only need to test the interpreting behaviour, moving this test to the interpreter unit tests under `runtime/tests/interpreter` (`indexing_test.go` file) might be a better option.

**Review by SupunS — COMMENTED** [review]
First contribution! 👏  Thank you for opening a PR to fix this!

Looks like there are some tests failing, due to the updated position info. Can you please update those tests as well? https://github.com/onflow/cadence/actions/runs/10775433087/job/29882238324?pr=3575
You can also run `make test` from the root directory of the repo, to run all tests locally.

File: runtime/parser/expression.go (hunk 1 above)
Comment (RZhang05) [reply runtime/parser/expression.go:1265]: `token` is now omitted.

File: runtime/account_test.go (hunk 2 above)
Comment (RZhang05) [reply runtime/account_test.go]: Given there are multiple equivalent tests in `runtime/tests/interpreter/interpreter_test.go` which cover the case of errors being thrown using the update index expression positions, this test has been removed.

Comment (SupunS) [reply runtime/account_test.go]: make sense 👍

**Review by SupunS — APPROVED** [review]
Nice work! 🎉

**Review by turbolent — APPROVED** [review]
Great work on your first PR! 👏

**Pull request merged**